services/src/performance-tests/src/test/groovy/org/openkilda/performancetests/spec/EnduranceV2Spec.groovy

```diff
@@ -0,0 +1,201 @@
+     * or just being idle.
+     * At the end of the test verify that all flows are valid and switches don't have any missing or excess entities.
+     * During the test all flows will be continuously 'pinged'. Any failed ping will be logged, any twice-failed ping
+     * will fail the test at the end
+     */
+    @Unroll
+    def "Simulate live environment with random events happening(#preset.name)"() {
+        Assume.assumeThat(preset.name, equalTo("local"))
+
+        setup: "Create a topology and a 'dice' with random events"
+        def topo = topoHelper.createRandomTopology(preset.switchesAmount, preset.islsAmount)
+        topoHelper.setTopology(topo)
+        flowHelper.setTopology(topo)
+        flowHelperV2.setTopology(topo)
+        def dice = new Dice([
+                new Face(chance: 25, event: this.&deleteFlow),
+                new Face(chance: 25, event: { createFlow(true) }),
+                new Face(chance: 25, event: { blinkIsl(topo.isls) }),
+                new Face(chance: 0, event: { TimeUnit.SECONDS.sleep(3) }),
+                new Face(chance: 25, event: { massReroute(topo) })
+        ])
+
+        and: "As starting point, create some amount of random flows in it"
+        preset.flowsToStartWith.times { createFlow() }
+        Wrappers.wait(flows.size() * 1.5) {
+            flows.each {
+                assert northbound.getFlowStatus(it.flowId).status == FlowState.UP
+                northbound.validateFlow(it.flowId).each { direction -> assert direction.asExpected }
+            }
+        }
+
+        when: "With certain probability one of the following events occurs: flow creation, flow deletion, isl blink, \
+idle, mass manual reroute. Step repeats pre-defined number of times"
+        def pinger = new FlowPinger(northbound, flows.collect { it.flowId }, rerouteDelay)
+        pinger.start()
+        preset.eventsAmount.times {
+            log.debug("running event #$it")
+            dice.roll()
+            TimeUnit.SECONDS.sleep(preset.pauseBetweenEvents)
+        }
+        def pingFailures = pinger.stop()
+
+        then: "All flows remain up and valid, with no missing rules on switches"
+        Wrappers.wait(WAIT_OFFSET * 2) {
+            northbound.getAllLinks().every { it.state == IslChangeType.DISCOVERED }
+        }
+        Wrappers.wait(WAIT_OFFSET + preset.switchesAmount) {
+            def soft = new SoftAssertions()
+            flows.each { flow ->
+                soft.checkSucceeds { assert northbound.getFlowStatus(flow.flowId).status == FlowState.UP }
+                soft.checkSucceeds {
+                    northbound.validateFlow(flow.flowId).each { direction -> assert direction.asExpected }
+                }
+            }
+            topo.switches.each { sw ->
+                def validation = northbound.validateSwitch(sw.dpId)
+                soft.checkSucceeds { assert validation.rules.missing.empty, sw }
```

Comment:
you can use helper method 
```
switchHelper.verifyRuleSectionsAreEmpty(validation, ["missing", "excess"])
switchHelper.verifyMeterSectionsAreEmpty(validation, ["missing", "misconfigured", "excess"])
```

Reply:
I will be missing the switch information. Since this is a 'soft' verification + there are 30+ switches in the system, it will be very difficult to track down exactly which switch failed to pass verification.